src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -48,7 +48,7 @@ class EnsembleFlowCustomNodePipelineExecutionTest : public TestWithTempDir {
         TestWithTempDir::SetUp();
 
         defaultContext = std::make_unique<ExecutionContext>(ExecutionContext::Interface::GRPC, ExecutionContext::Method::Predict);
-        reporter = std::make_unique<ModelMetricReporter>(nullptr, "example_pipeline_name", 1);  // TODO: Add real metric registry
+        reporter = std::make_unique<ModelMetricReporter>(nullptr, nullptr, "example_pipeline_name", 1);  // TODO: Add real metric registry
```

Comment:
My new commits passed real MetricRegistry here.
This is to make sure the metrics are getting incremented and test pass with no error.
Do the same with MetricConfig? To make it easier, you could define default constructor for MetricConfig with standard setting like - enabled=True.

Reply:
I synced with your branch. Let's leave nullptr for now.